openstudiocore/src/bimserver/ProjectImportation.cpp

```diff
@@ -0,0 +1,347 @@
+*
+*  This library is distributed in the hope that it will be useful,
+*  but WITHOUT ANY WARRANTY; without even the implied warranty of
+*  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+*  Lesser General Public License for more details.
+*
+*  You should have received a copy of the GNU Lesser General Public
+*  License along with this library; if not, write to the Free Software
+*  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+**********************************************************************/
+
+#include "ProjectImportation.hpp"
+#include "BIMserverConnection.hpp"
+
+#include "../osversion/VersionTranslator.hpp"
+
+
+#include <QFileDialog>
+#include <QGridLayout>
+#include <QVBoxLayout>
+#include <QMessageBox>
+#include <QLabel>
+#include <QEventLoop>
+
+#include <QLineEdit>
+#include <QPushButton>
+#include <QDebug>
+
+namespace openstudio {
+namespace bimserver {
+
+	ProjectImportation::ProjectImportation(QWidget *parent) :
+		QDialog(parent) {
+		QGridLayout *subLayout = new QGridLayout;
+		QVBoxLayout *mainLayout = new QVBoxLayout;
+
+		QLabel *introLabel = new QLabel("Please select a project to import: ", this);
+		m_proList = new QListWidget(this);
+		m_ifcList = new QListWidget(this);
+		QPushButton *okButton = new QPushButton(tr("Download OSM File"), this);
+		QPushButton *newButton = new QPushButton(tr("New Project"), this);
+		QPushButton *loadButton = new QPushButton(tr("Check in IFC File"), this);
+		QPushButton *selectButton = new QPushButton(tr(" > "), this);
+		QPushButton *cancelButton = new QPushButton(tr("Cancel"), this);
+		QPushButton *settingButton = new QPushButton(tr("Setting"), this);
+		m_statusBar = new QStatusBar(this);
+		
+		connect(okButton, SIGNAL(clicked()),this, SLOT(okButton_clicked()));
+		connect(newButton, SIGNAL(clicked()),this, SLOT(newButton_clicked()));
+		connect(loadButton, SIGNAL(clicked()), this, SLOT(loadButton_clicked()));
+		connect(selectButton, SIGNAL(clicked()), this, SLOT(selectButton_clicked()));
+		connect(settingButton, SIGNAL(clicked()),this, SLOT(settingButton_clicked()));
+		connect(cancelButton, SIGNAL(clicked()), this, SLOT(close()));
+
+		subLayout->addWidget(m_proList, 1, 0, 1, 1);
+		subLayout->addWidget(m_ifcList, 1, 2, 1, 1);
+
+		subLayout->addWidget(introLabel, 0, 0, 1, 3);
+		subLayout->addWidget(selectButton, 1, 1, 1, 1);
+		subLayout->addWidget(newButton, 2, 0, 1, 1);
+		subLayout->addWidget(loadButton, 2, 2, 1, 1);
+		subLayout->addWidget(okButton, 3, 0, 1, 1);
+		subLayout->addWidget(settingButton, 3, 1, 1, 1);
+		subLayout->addWidget(cancelButton, 3, 2, 1, 1);
+
+		mainLayout->addLayout(subLayout);
+
+		mainLayout->addWidget(m_statusBar);
+
+		setLayout(mainLayout);
+
+		setWindowTitle("Import Project");
+
+		m_waitForOSM = new QEventLoop(this);
+	}
+
+	ProjectImportation::~ProjectImportation() {
+
+	}
+
+	boost::optional<model::Model> ProjectImportation::run() {
```

Comment:
@zhouchong90 in the future it might be nice to factor out a way to call this without requiring human interaction (e.g. through a script or something).  Not needed for now